### PR TITLE
Date time fixes for 32 bit systems and valgrind

### PR DESCRIPTION
**Pull request by daviditen**
Fix compile time problem in DateTime module for 32 bit systems.  A 64 bit 'int' was assigned to a 'c_long', which is a 32 bit value on our 32 bit systems.  Add a cast to 'c_long'.  This gets the testTime.chpl test to compile on these systems, but it is still tripping an assert in the test due to an apparent denormalize issue.

Initialize all fields of an extern struct before passing it to an extern function. Valgrind was giving errors about uninitialized values otherwise.

### CODE REVIEW

**Comment by ben-albrecht** [comment]
Do you have more info on the denormalize issue, i.e. is it documented elsewhere?

**Comment by daviditen** [comment]
I don't have more info about it. I just discovered that the failures would come and go after small changes like adding writeln calls to try to look at the values the asserts fail for, so tried --no-denormalize and found it worked.  I still need to investigate that further.

File: modules/standard/DateTime.chpl [hunk 1]
```diff
@@ -613,12 +620,15 @@ module DateTime {
     timeStruct.tm_sec = second: int(32);
     timeStruct.tm_min = minute: int(32);
     timeStruct.tm_hour = hour: int(32);
-    timeStruct.tm_year = 1900;
+    timeStruct.tm_year = 0;
     timeStruct.tm_mday = 1;
     timeStruct.tm_mon = 1;
 
+    timeStruct.tm_wday = ((new date(1900, 1, 1)).weekday():int(32) + 1) % 7;
```
Comment (ben-albrecht) [inline modules/standard/DateTime.chpl]: Was this previously occurring automatically as a result of `timeStruct.tm_year = 1900` ?

Comment (daviditen) [inline modules/standard/DateTime.chpl]: No, it was uninitialized before.  timeStruct.tm_year = 1900 was a mistake - the year is supposed to be 1900 based, so 0 is 1900.

**Comment by daviditen** [comment]
Thanks @ben-albrecht !

**Pull request merged**